pkg/controller/infinispan/util/security.go

```diff
@@ -75,7 +78,11 @@ func FindPassword(usr string, descriptor []byte) (string, error) {
 }
 
 // GetSecretName returns the secret name associated with a server
-func GetSecretName(name string) string {
-	return fmt.Sprintf("%v-generated-secret", name)
+func GetSecretName(m *infinispanv1.Infinispan) string {
+	if m.Spec.Security.EndpointSecretName == "" {
+		return fmt.Sprintf("%v-generated-secret", m.GetName())
+	} else {
```

Comment:
go-lint suggests that the else keyword is not needed (due to the return in the if condition)

Reply:
Thanks, @rigazilla , will update.